.github/get-changed-adapters-array.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# This is a utility script to get a list of changed adapters. You can use it in a helper script with the following:
+# source .github/get-changed-adapters-array.sh
+# # Iteration:
+#     CHANGED_ADAPTERS=$(getChangedAdapterArray)
+#     for adapter in $CHANGED_ADAPTERS; do
+#       echo "$adapter"
+#     done
+# # As space separated string:
+#     SPACE_SEPARATED=$(getChangedAdapterArray)
+
+
+if [[ -z $UPSTREAM_BRANCH ]]; then
+  UPSTREAM_BRANCH=develop
+fi
+REGEX="@chainlink/(.*-adapter)"
+
+# DO NOT USE echo IN THIS SCRIPT EXCEPT TO DUMP THE RESULT
+# Any echoed lines will appear in the output array.
+function getChangedAdapterArray(){
+  adapterSet=()
+  for line in $(git diff --name-only "origin/$UPSTREAM_BRANCH"...HEAD); do
+    # Returning this keyword will instruct the workflow to run against all adapters instead of a filtered list
+    if [[ $line =~ ^packages/(core|scripts|non-deployable) ||  $line =~ grafana/ ]]; then
```

Comment:
Just realized that this would not account for the case where a source adapter changes, and the composite adapter imports that one from legos (which is something I really dislike). I don't know how easy it would be to update this script to check for this, so maybe we always include composites and the optimization on sources would still be really noticeable